Failed to apply plugin with 'listenablefuture' library on the classpath

I'm applying the plugins via `buildSrc`:

```kotlin
plugins {
    java
    id("org.gradlex.jvm-dependency-conflict-resolution")
    id("org.gradlex.jvm-dependency-conflict-detection")
}
```
(version `2.0`)

Then I apply the convention to my root project.

When I try to assemble the project I receive the next exception:
```kotlin
     ...
Caused by: org.gradle.api.InvalidUserDataException: Cannot add capability com.google.guava:listenablefuture with version 1.0 because it's already defined with version 9999.0-empty-to-avoid-conflict-with-guava
	at org.gradle.internal.component.external.model.DefaultMutableCapabilitiesMetadata.addCapability(DefaultMutableCapabilitiesMetadata.java:45)
	at org.gradlex.jvm.dependency.conflict.detection.rules.CapabilityDefinitionRule.lambda$execute$0(CapabilityDefinitionRule.java:41)
	at org.gradle.internal.component.external.model.VariantMetadataRules$VariantAction.maybeExecute(VariantMetadataRules.java:213)
	at org.gradle.internal.component.external.model.VariantMetadataRules.applyCapabilitiesRules(VariantMetadataRules.java:110)
	at org.gradle.internal.component.external.model.DefaultConfigurationMetadata.getCapabilities(DefaultConfigurationMetadata.java:143)
	at org.gradle.internal.component.external.model.maven.RealisedMavenModuleResolveMetadata.transform(RealisedMavenModuleResolveMetadata.java:97)
	at org.gradle.api.internal.artifacts.dsl.DefaultComponentMetadataProcessor.realizeMetadata(DefaultComponentMetadataProcessor.java:89)
	at org.gradle.api.internal.artifacts.dsl.DefaultComponentMetadataProcessor.lambda$static$0(DefaultComponentMetadataProcessor.java:74)
	at org.gradle.internal.resolve.caching.CrossBuildCachingRuleExecutor.executeRule(CrossBuildCachingRuleExecutor.java:192)
	at org.gradle.internal.resolve.caching.CrossBuildCachingRuleExecutor.tryFromCache(CrossBuildCachingRuleExecutor.java:139)
	at org.gradle.internal.resolve.caching.CrossBuildCachingRuleExecutor.execute(CrossBuildCachingRuleExecutor.java:110)
	at org.gradle.api.internal.artifacts.dsl.DefaultComponentMetadataProcessor.processClassRuleWithCaching(DefaultComponentMetadataProcessor.java:241)
	... 204 more
```

I've tried different workarounds but no luck, please help

--- a/src/main/java/org/gradlex/jvm/dependency/conflict/detection/rules/CapabilityDefinitionRule.java
+++ b/src/main/java/org/gradlex/jvm/dependency/conflict/detection/rules/CapabilityDefinitionRule.java
@@ -38,9 +38,16 @@ public abstract class CapabilityDefinitionRule implements ComponentMetadataRule 
     public final void execute(ComponentMetadataContext context) {
         if (shouldApply(context.getDetails().getId())) {
             context.getDetails().allVariants(variant -> {
-                variant.withCapabilities(capabilities -> capabilities.addCapability(
-                        definition.getGroup(), definition.getCapabilityName(), getVersion(context.getDetails().getId())
-                ));
+                variant.withCapabilities(capabilities -> {
+                    // remove capability if it already exists so that it can be added back
+                    // with a potentially different version
+                    capabilities.removeCapability(
+                            definition.getGroup(), definition.getCapabilityName()
+                    );
+                    capabilities.addCapability(
+                            definition.getGroup(), definition.getCapabilityName(), getVersion(context.getDetails().getId())
+                    );
+                });
                 additionalAdjustments(variant);
             });
         }

--- a/src/test/groovy/org/gradlex/jvm/dependency/conflict/test/CapabilityWithDifferentVersionsTest.groovy
+++ b/src/test/groovy/org/gradlex/jvm/dependency/conflict/test/CapabilityWithDifferentVersionsTest.groovy
@@ -3,6 +3,8 @@ package org.gradlex.jvm.dependency.conflict.test
 import org.gradlex.jvm.dependency.conflict.test.fixture.GradleBuild
 import spock.lang.Specification
 
+import static org.gradlex.jvm.dependency.conflict.test.fixture.GradleBuild.GRADLE6_TEST
+
 class CapabilityWithDifferentVersionsTest extends Specification {
 
     @Delegate
@@ -37,8 +39,44 @@ class CapabilityWithDifferentVersionsTest extends Specification {
                 println(configurations.compileClasspath.get().files.joinToString("\\n") { it.name })
             }
         """
+        if (GRADLE6_TEST) { configureEnvAttribute() }
 
         expect:
         printJars()
+    }
+
+    def "does not fail with empty listenable future dependency on the classpath"() {
+        given:
+        buildFile << """
+            plugins {
+                id("org.gradlex.jvm-dependency-conflict-resolution")
+                id("java-library")
+            }
+            tasks.withType<JavaCompile>().configureEach {
+                options.release.set(17)
+            }
+            repositories.mavenCentral()
+            dependencies {
+                 implementation(platform("com.google.cloud:spring-cloud-gcp-dependencies:5.2.0"))
+                 implementation("com.google.cloud:spring-cloud-gcp-starter-bigquery")
+            }
+
+            tasks.register("printJars") {
+                println(configurations.compileClasspath.get().files.joinToString("\\n") { it.name })
+            }
+        """
+        if (GRADLE6_TEST) { configureEnvAttribute() }
+
+        expect:
+        printJars()
+    }
+
+    void configureEnvAttribute() {
+        buildFile << """
+            val envAttribute = Attribute.of("org.gradle.jvm.environment", String::class.java)
+            configurations.compileClasspath {
+                attributes.attribute(envAttribute, "standard-jvm")
+            }
+        """
     }
 }

--- a/src/test/groovy/org/gradlex/jvm/dependency/conflict/test/CapabilityWithDifferentVersionsTest.groovy
+++ b/src/test/groovy/org/gradlex/jvm/dependency/conflict/test/CapabilityWithDifferentVersionsTest.groovy
@@ -1,0 +1,44 @@
+package org.gradlex.jvm.dependency.conflict.test
+
+import org.gradlex.jvm.dependency.conflict.test.fixture.GradleBuild
+import spock.lang.Specification
+
+class CapabilityWithDifferentVersionsTest extends Specification {
+
+    @Delegate
+    GradleBuild build = new GradleBuild()
+
+    def "does not fail if capability is added several times with different versions"() {
+        given:
+        buildFile << """
+            plugins {
+                id("org.gradlex.jvm-dependency-conflict-detection") apply false
+                id("java-library")
+            }
+            repositories.mavenCentral()
+            dependencies.components {
+                withModule("com.google.guava:guava") {
+                    allVariants {
+                        withCapabilities {
+                            addCapability("com.google.guava", "listenablefuture", 
+                                "9999.0-empty-to-avoid-conflict-with-guava")
+                        }
+                    }
+                }
+            }
+            
+            apply(plugin = "org.gradlex.jvm-dependency-conflict-detection")
+            
+            dependencies {
+                api("com.google.guava:guava:33.0.0-jre")
+            }
+
+            tasks.register("printJars") {
+                println(configurations.compileClasspath.get().files.joinToString("\\n") { it.name })
+            }
+        """
+
+        expect:
+        printJars()
+    }
+}

--- a/src/test/groovy/org/gradlex/jvm/dependency/conflict/test/GuavaClasspathTest.groovy
+++ b/src/test/groovy/org/gradlex/jvm/dependency/conflict/test/GuavaClasspathTest.groovy
@@ -17,6 +17,16 @@ class GuavaClasspathTest extends Specification {
 
     static allGuavaVersions() {
         [
+                ['33.2.0', 'jre'    , [errorProne:  '2.26.1', j2objc: '3.0.0', jsr305: '3.0.2', checker: '3.42.0', failureaccess: '1.0.2']],
+                ['33.2.0', 'android', [errorProne:  '2.26.1', j2objc: '3.0.0', jsr305: '3.0.2', checker: '3.42.0', failureaccess: '1.0.2']],
+                ['33.1.0', 'jre'    , [errorProne:  '2.26.1', j2objc: '3.0.0', jsr305: '3.0.2', checker: '3.42.0', failureaccess: '1.0.2']],
+                ['33.1.0', 'android', [errorProne:  '2.26.1', j2objc: '3.0.0', jsr305: '3.0.2', checker: '3.42.0', failureaccess: '1.0.2']],
+                ['33.0.0', 'jre'    , [errorProne:  '2.23.0', j2objc: '2.8', jsr305: '3.0.2', checker: '3.41.0', failureaccess: '1.0.2']],
+                ['33.0.0', 'android', [errorProne:  '2.23.0', j2objc: '2.8', jsr305: '3.0.2', checker: '3.41.0', failureaccess: '1.0.2']],
+                ['32.1.3', 'jre'    , [errorProne:  '2.21.1', j2objc: '2.8', jsr305: '3.0.2', checker: '3.37.0', failureaccess: '1.0.1']],
+                ['32.1.3', 'android', [errorProne:  '2.21.1', j2objc: '2.8', jsr305: '3.0.2', checker: '3.37.0', failureaccess: '1.0.1']],
+                ['32.1.2', 'jre'    , [errorProne:  '2.18.0', j2objc: '2.8', jsr305: '3.0.2', checker: '3.33.0', failureaccess: '1.0.1']],
+                ['32.1.2', 'android', [errorProne:  '2.18.0', j2objc: '2.8', jsr305: '3.0.2', checker: '3.33.0', failureaccess: '1.0.1']],
                 ['32.1.1', 'jre'    , [errorProne:  '2.18.0', j2objc: '2.8', jsr305: '3.0.2', checker: '3.33.0', failureaccess: '1.0.1']],
                 ['32.1.1', 'android', [errorProne:  '2.18.0', j2objc: '2.8', jsr305: '3.0.2', checker: '3.33.0', failureaccess: '1.0.1']],
                 ['32.0.1', 'jre'    , [errorProne:  '2.18.0', j2objc: '2.8', jsr305: '3.0.2', checker: '3.33.0', failureaccess: '1.0.1']],
@@ -111,12 +121,12 @@ class GuavaClasspathTest extends Specification {
                 result.add([it[0], it[1], it[2], 'standard-jvm', 'runtimeClasspath'])
             }
         }
-        if (System.getProperty("gradleVersionUnderTest") == "7.5.1") {
+        if (System.getProperty("gradleVersionUnderTest") == "7.6.4") {
             // only do all permutations for one Gradle version
             return result
         }
         // reduced amount of permutations
-        return result.subList(0, 32)
+        return result.subList(0, 80)
     }
 
     @Unroll
@@ -160,7 +170,7 @@ class GuavaClasspathTest extends Specification {
         """
 
         expect:
-        expectedClasspath(guavaVersion, jvmEnv, classpath, dependencyVersions) == printJars().output.split('\n') as Set
+        expectedClasspath(guavaVersion, jvmEnv, classpath, dependencyVersions) == printJars().output.split('\n') as TreeSet
 
         where:
         [guavaVersion, versionSuffix, dependencyVersions, jvmEnv, classpath] << allGuavaCombinations(true)
@@ -169,7 +179,7 @@ class GuavaClasspathTest extends Specification {
     Set<String> expectedClasspath(String guavaVersion, String jvmEnv, String classpath, Map<String, String> dependencyVersions) {
         int majorGuavaVersion = guavaVersion.substring(0, 2) as Integer
         String jarSuffix = majorGuavaVersion < 22 ? '' : jvmEnv == 'android' ? 'android' : (guavaVersion == '22.0' || guavaVersion == '23.0') ? '' : 'jre'
-        Set<String> result = ["guava-${guavaVersion}${jarSuffix? '-' : ''}${jarSuffix}.jar"]
+        Set<String> result = ["guava-${guavaVersion}${jarSuffix? '-' : ''}${jarSuffix}.jar"] as TreeSet
         if (dependencyVersions.failureaccess) {
             result += "failureaccess-${dependencyVersions.failureaccess}.jar"
         }

